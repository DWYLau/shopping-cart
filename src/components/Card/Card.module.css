.card {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  min-height: 52vh;
  width: 85%;
  background-color: #f8f8f4;
  border: 1px solid #dcdcdc;
  border-radius: 5px;
}

.card h3 {
  text-align: center;
  width: 70%;
  font-family: system-ui, sans-serif;
  font-size: 1.1rem;
  font-weight: bold;
  line-height: 1.1;
}

.image {
  background-color: white;
  padding: 1rem;
  height: 25vh;
  width: 10vw;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
}

.rating-container {
  display: flex;
  gap: 1rem;
}

.rating,
.likes {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 0.3rem;
}

.icon {
  height: 1rem;
}

.button-container {
  display: flex;
  gap: 0.5rem;
  margin-left: 2rem;
}

.cart-button {
  cursor: pointer;
  padding: 0.7rem;
  font-family: system-ui, sans-serif;
  font-weight: bold;
  line-height: 1.1;
  border: 2px solid #f8f8f4;
  border-radius: 8px;
  background-color: #ccff00;
  min-height: 100%;
}

.cart-button:hover {
  border: 2px solid #5f9ea0;
}

.input-number {
  border: none;
  background-color: #d3d3d3;
  border: 2px solid #f8f8f4;
  border-radius: 8px;
  font-size: 1rem;
  padding: 0.4rem;
  min-height: 100%;
  width: 23%;
}

.input-number:hover {
  border: 2px solid #5f9ea0;
  background-color: #b2beb5;
}
