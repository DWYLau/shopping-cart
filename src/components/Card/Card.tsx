import styles from "./Card.module.css"
import star from "../../assets/icons/star.png"
import like from "../../assets/icons/like.png"
import { Product } from "../../utils/types"

interface Props {
  product: Product
  handleQuantity: (event: React.ChangeEvent<HTMLInputElement>) => void
  addCart: (product: Product) => void
}

function Card({ product, handleQuantity, addCart }: Props) {
  return (
    <div key={product.id} className={styles.card}>
      <img className={styles.image} src={product.image} alt='' />
      <h3>{product.title}</h3>

      <div className={styles["rating-container"]}>
        <div className={styles.rating}>
          <img className={styles.icon} src={star} alt='Rating' />
          <h3>{product.rating.rate}</h3>
        </div>

        <div className={styles.likes}>
          <img className={styles.icon} src={like} alt='Likes' />
          <h3>{product.rating.count}</h3>
        </div>

        <h3>Â£{product.price}</h3>
      </div>

      <div className={styles["button-container"]}>
        <input
          onChange={handleQuantity}
          type='number'
          min={0}
          max={100}
          defaultValue={0}
          className={styles["input-number"]}
        />
        <button
          onClick={() => addCart(product)}
          className={styles["cart-button"]}
        >
          Add To Cart
        </button>
      </div>
    </div>
  )
}

export default Card
