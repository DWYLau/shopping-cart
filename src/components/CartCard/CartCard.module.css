.card {
  display: flex;
  justify-content: space-between;
  padding: 1rem;
  border-bottom: 2px solid #c0c0c0;
  font-family: system-ui, sans-serif;
  font-size: 1.5rem;
}

.image {
  background-color: white;
  padding: 1rem;
  height: 25vh;
  width: 10vw;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
}

.summary {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  width: 27vw;
  gap: 1rem;
  margin-left: 2rem;
}

.extra-info {
  display: flex;
  gap: 1rem;
}

.quantity {
  padding-right: 1rem;
  border-right: 2px solid #c0c0c0;
}

.button-container {
  display: flex;
  gap: 1rem;
}

.button {
  cursor: pointer;
  padding: 1rem;
  font-family: system-ui, sans-serif;
  font-size: 1.2rem;
  font-weight: bold;
  line-height: 1.1;
  border: 2px solid #f8f8f4;
  border-radius: 8px;
  background-color: #ccff00;
  min-height: 100%;
}

.button:hover {
  border: 2px solid #5f9ea0;
}

.total {
  display: flex;
  flex-direction: column;
  gap: 1rem;
  margin-left: 3rem;
  margin-top: 1.5rem;
}
