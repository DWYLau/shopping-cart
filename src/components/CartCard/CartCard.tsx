import { Product } from "../../utils/types"
import styles from "./CartCard.module.css"

interface Props {
  product: Product
}

function CartCard({ product }: Props) {
  return (
    <div className={styles.card}>
      <img className={styles.image} src={product.image} alt="Product Image" />
      <div className={styles.summary}>
        <h3>{product.title}</h3>
        <div className={styles["extra-info"]}>
          <h3 className={styles.quantity}>Quantity: {product.number}</h3>
          <h3>Price: £{product.price.toFixed(2)}</h3>
        </div>
        <div className={styles["button-container"]}>
          <button className={styles.button}>Change Quantity</button>
          <button className={styles.button}>Remove</button>
        </div>
      </div>
      <div className={styles.total}>
        <h2>Total Price</h2>
        <h3>£{product.total?.toFixed(2)}</h3>
      </div>
    </div>
  )
}

export default CartCard
