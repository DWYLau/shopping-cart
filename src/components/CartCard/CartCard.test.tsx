import CartCard from "./CartCard"
import { BrowserRouter } from "react-router-dom"
import { describe, it, expect } from "vitest"
import { getByText, render, screen } from "@testing-library/react"
import { product } from "../Card/data"
import userEvent from "@testing-library/user-event"

describe("Cart Card component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <CartCard product={product} />
      </BrowserRouter>
    )

    expect(container).toMatchSnapshot()
  })

  it("renders product title", () => {
    const { container } = render(
      <BrowserRouter>
        <CartCard product={product} />
      </BrowserRouter>
    )

    getByText(
      container,
      "Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops"
    )
  })

  it("change quantity button is clicked", async () => {
    const user = userEvent.setup()

    render(
      <BrowserRouter>
        <CartCard product={product} />
      </BrowserRouter>
    )

    const button = screen.getByRole("button", {
      name: "Change Quantity",
    }) as HTMLElement

    await user.click(button)

    expect(screen.getByText("Confirm")).toBeTruthy()
  })
})
