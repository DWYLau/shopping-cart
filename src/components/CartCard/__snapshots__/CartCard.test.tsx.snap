// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cart Card component > renders correctly 1`] = `
<div>
  <div
    class="_card_6295e2"
  >
    <img
      alt="Product Image"
      class="_image_6295e2"
      src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
    />
    <div
      class="_summary_6295e2"
    >
      <h3>
        Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
      </h3>
      <div
        class="_extra-info_6295e2"
      >
        <h3
          class="_quantity_6295e2"
        >
          Quantity: 
        </h3>
        <h3>
          Price: £
          109.95
        </h3>
      </div>
      <div
        class="_button-container_6295e2"
      >
        <button
          class="_button_6295e2"
        >
          Change Quantity
        </button>
        <button
          class="_button_6295e2"
        >
          Remove
        </button>
      </div>
    </div>
    <div
      class="_total_6295e2"
    >
      <h2>
        Total Price
      </h2>
      <h3>
        £
      </h3>
    </div>
  </div>
</div>
`;
